Modules/posixmodule.c

```diff
@@ -2712,13 +2712,14 @@ static PyObject *
 os_ttyname_impl(PyObject *module, int fd)
 /*[clinic end generated code: output=c424d2e9d1cd636a input=9ff5a58b08115c55]*/
 {
-    char *ret;
+    int ret;
+    char buffer[MAXPATHLEN];
```

Comment:
seems like the right buffer size is actually `TTY_NAME_MAX`

Reply:
Correct. This macro wasn't mentioned in the version of the man pages of my local dev machine (i.e: `man ttyname`) but after googling this constant, it *is indeed* mentioned in `man limits.h` which is imported by `Python.h` in the base code.
Good catch! Thanks for the recommendation!